core/test/unit/middleware/cors_spec.js

```diff
@@ -137,4 +141,41 @@ describe('cors', function () {
 
         done();
     });
+
+    it('should be enabled if the origin matches config.url', function (done) {
+        var origin = 'http://my.blog';
+        configUtils.set({
+            url: origin
+        });
+
+        req.get = sinon.stub().withArgs('origin').returns(origin);
+        res.get = sinon.stub().withArgs('origin').returns(origin);
+        req.headers.origin = origin;
+
+        cors(req, res, next);
+
+        next.called.should.be.true();
+        res.headers['Access-Control-Allow-Origin'].should.equal(origin);
+
+        done();
+    });
```

Comment:
As far as I can tell, the issue here then is that the `Origin` matching `config.url` is not an indicator that this is a same-origin request. The request destination may be `config.urlSSL` or any other hostname that your server is configured to respond to. 

In this instance, `config.url` is effectively a default "trusted domain" for the `ghost-frontend` client. So perhaps that's a better way to code it in?